cmake/GenerateVersion.cmake

```diff
@@ -8,6 +8,17 @@ execute_process(
   ERROR_QUIET
   RESULT_VARIABLE RES)
 
+execute_process(
```

Comment:
Can't we use `NVIM_VERSION_HASH` (provided below)?

Reply:
oh I meant `NVIM_VERSION_GIT` or whatever generates the current version string. But I see that `NVIM_VERSION_GIT` has a value like `145+g0f9113907` which isn't quite what we want.

So I think `git rev-parse` as you've done is needed after all...

Reply:
Just for the sake of clarity, want me to change it to `NVIM_VERSION_BUILD` prefixed with a `g`?

Reply:
re-using `NVIM_VERSION_GIT` is probably less code anyway, besides also being more robust. (see edit above)

Reply:
Sorry, wasn't paying attention meant to say `NVIM_VERSION_GIT`, not `NVIM_VERSION_BUILD`. I'll make the change to use `NVIM_VERSION_GIT` real quick.

Reply:
> Another note on using NVIM_VERSION_HASH is that it seemingly changes on every build regardless of file modifications (weird!)

Because you're changing (appending) the file on every build